mantidimaging/core/operations/rescale/rescale.py

```diff
@@ -42,9 +42,15 @@ def filter_func(images: ImageStack,
         :return: The ImageStack object scaled to a new range.
         """
 
-        RescaleFilter.filter_array(images.data, min_input, max_input, max_output)
+        params = {'min_input': min_input, 'max_input': max_input, 'max_output': max_output}
+        ps.run_compute_func(RescaleFilter.compute_function, len(images.data), [images.shared_array], params)
         return images
 
+    @staticmethod
+    def compute_function(index: int, array: np.ndarray, params: dict):
+        min_input, max_input, max_output = params['min_input'], params['max_input'], params['max_output']
+        array[index] = np.interp(array[index], [min_input, max_input], [0, max_output])
+
     @staticmethod
     def filter_array(image: np.ndarray, min_input: float, max_input: float, max_output: float) -> np.ndarray:
         image[:] = np.interp(image, [min_input, max_input], [0, max_output])
```

Comment:
It looks like we can't remove `filter_array()` because it gets used in a few other places in the code. So it would be good to call `filter_array()` from `compute_function()` to avoid duplication.

Something like
```python
    def compute_function(index: int, array: np.ndarray, params: dict):
        min_input, max_input, max_output = params['min_input'], params['max_input'], params['max_output']
        array[index] = filter_array(array[index], min_input, ...

```